mio/process/video.py

```diff
@@ -156,13 +156,27 @@ def process_frame(self, input_frame: np.ndarray) -> Optional[np.ndarray]:
         if input_frame is None:
             return None
 
-        if self.noise_patch_config.enable and self.previous_frame is not None:
-            # Call the unified detection method
-            broken, noise_patch = self.noise_detect_helper.detect_frame_with_noisy_buffer(
-                input_frame,
-                self.previous_frame if self.noise_patch_config.method == "mean_error" else None,
-                self.noise_patch_config,
-            )
+        if self.noise_patch_config.enable:
```

Comment:
to avoid having this entire method gaining an extra layer of indentation and and more layers of `if/else` switches, this should probably be 

```python
if not self.noise_patch_config.enable:
    return input_frame
```